contracts/governance/extensions/GovernorVotingSimple.sol

```diff
@@ -19,63 +19,77 @@ abstract contract GovernorVotingSimple is IGovernor {
         Abstain
     }
 
-    struct Voting {
+    struct Receipt {
         uint256 againstVotes;
         uint256 forVotes;
         uint256 abstainVotes;
-        mapping (address => bool) hasVoted;
+        mapping(address => bool) hasVoted;
     }
 
-    mapping (uint256 => Voting) private _votings;
+    mapping(uint256 => Receipt) private _receipts;
 
     /**
      * @dev See {IGovernor-hasVoted}.
      */
     function hasVoted(uint256 proposalId, address account) public view virtual override returns (bool) {
-        return _votings[proposalId].hasVoted[account];
+        return _receipts[proposalId].hasVoted[account];
     }
 
     /**
-     * @dev See {IGovernor-proposalWeight}.
+     * @dev Accessor to the internal vote counts.
      */
-    function proposalWeight(uint256 proposalId) public view virtual override returns (uint256) {
-        return _votings[proposalId].againstVotes
-            + _votings[proposalId].forVotes
-            + _votings[proposalId].abstainVotes;
+    function proposalVotes(uint256 proposalId)
```

Comment:
This function is not a part of any interface. Do we have a use case in mind? I'm considering alternatives like `proposalVotes(uint256 proposalId, uint8 support)` to make it more generic so that other voting modules could define it as well, and that it could be declared in the core.

Reply:
honestly, I mostly needed it for testing, and I noticed their was no way to access this data (including from a inheriting contract like a mock) if an accessor was not available.

I thought that having some access would be great ... but yes, it felt module-specific. Not sure how to make it generic. Will think about it.

Reply:
What about `proposalVotes(uint256 proposalId, uint8 support) returns (uint256)`?

Reply:
I can imagine module were this wouldn't make sense/be possible to return.


```
abstract contract GovernorCountingScore is Governor {
    struct Voting {
        uint256 supply;
        uint256 score;
        mapping (address => bool) hasVoted;
    }

    mapping (uint256 => Voting) private _votings;
    
    function maxScore() public view virtual returns (uint8) {
        return 100;
    }

    function requiredScore() public view virtual returns (uint8) {
        return 50;
    }

    function proposalWeight(uint256 proposalId) public view virtual override returns (uint256) {
        return _votings[proposalId].supply;
    }

    function proposalScore(uint256 proposalId) public view virtual returns (uint256) {
        return _votings[proposalId].score;
    }

    function hasVoted(uint256 proposalId, address account) public view virtual override returns (bool) {
        return _votings[proposalId].hasVoted[account];
    }

    function _quorumReached(uint256 proposalId) internal view virtual override returns (bool) {
        ProposalVote storage proposalvote = _proposalVotes[proposalId];

        return quorum(proposalSnapshot(proposalId)) < _votings[proposalId].supply;
    }

    function _voteSucceeded(uint256 proposalId) internal view virtual override returns (bool) {
        return _votings[proposalId].supply.requiredScore() < _votings[proposalId].score;
    }

    function _countVote(
        uint256 proposalId,
        address account,
        uint8 support,
        uint256 weight
    ) internal virtual override {
        require(support <= maxScore(), "ScoreVoting: voting over the maximum score");

        Voting storage voting = _votings[proposalId];

        require(!voting.hasVoted[account], "GovernorVotingSimple: vote already casted");
        voting.hasVoted[account] = true;
        voting.supply += weight;
        voting.score  += weight * support;
    }
}
```